File: 605-can-place-flowers/can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) 
    {
        int count1 = n;
        for(int i = 0;i < flowerbed.size();i++)
        {
            if((flowerbed[i] == 0) && (i == 0 || flowerbed[i-1] == 0) && (i == flowerbed.size()-1 || flowerbed[i+1] == 0))
            {
                count1--;
                ++i;
            }
        }
        if(count1 > 0)
        {
            return false;
        }
        return true; 
    }
};

File: 605-can-place-flowers/can-place-flowers.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        i = 0
        while i < len(flowerbed):
            if (flowerbed[i] == 0 and 
                (i == 0 or flowerbed[i - 1] == 0) and 
                (i == len(flowerbed) - 1 or flowerbed[i + 1] == 0)):

                flowerbed[i] = 1
                n -= 1
                i += 2
            else:
                i += 1
        return n <= 0